src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,136 @@
+    List<String> inputArray;
+
+    public Parser(String command) {
+        this.inputArray = Arrays.asList(command.split(" "));
+    }
+
+    /**
+     * Returns the first word in the input command.
+     *
+     * @return the keyword as a Command.
+     * @throws DukeException if the keyword is not recognised.
+     */
+    public Command getKeyword() throws DukeException {
+        try {
+            Command keyword = Command.valueOf(inputArray.get(0).toUpperCase());
+            return keyword;
+        } catch (Exception e) {
+            throw new DukeException("You did not provide a valid command");
+        }
+    }
+
+    /**
+     * Returns everything except the first keyword of the input command.
+     *
+     * @return the main content of the input.
+     * @throws DukeException if the input is incomplete.
+     */
+    public String getContent() throws DukeException {
+        try {
+            String content = String.join(" ", inputArray.subList(1, inputArray.size()));
+            return content;
+        } catch (Exception e) {
+            throw new DukeException("Input given is incomplete");
+        }
+    }
+
+    /**
+     * Returns content from an event command.
+     *
+     * @return the main content of the input.
+     * @throws DukeException if the input is incomplete.
+     */
+    public String[] getContentForEvent() throws DukeException {
+        String content = getContent();
+        if (!content.contains("/at")) {
+            throw new DukeException("An event must at least have a description and date!");
+        }
+        String[] commandArray = content.split(" /at ");
+        return commandArray;
+    }
+
+    /**
+     * Returns content from a deadline command.
+     *
+     * @return the main content of the input.
+     * @throws DukeException if the input is incomplete.
+     */
+    public String[] getContentForDeadline() throws DukeException {
+        String content = getContent();
+        if (!content.contains("/by")) {
+            throw new DukeException("A deadline must at least have a description and date!");
+        }
+        String[] commandArray = content.split(" /by ");
+        return commandArray;
+    }
+
+    static public LocalDateTime stringToDateTime(String str) throws DukeException {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            LocalDateTime dateTime = LocalDateTime.parse(str, formatter);
+            return dateTime;
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Date provided is invalid");
+        }
+    }
+
+    /**
+     * Parsed a saved string and returns the corresponding Task.
+     *
+     * @param save String containing the line in the storage to be parsed.
+     * @return the parsed Task.
+     * @throws DukeException if the saved string does not match any patterns.
+     */
+    static public Task parseSave(String save) throws DukeException {
```

Comment:
Maybe you can change the variable name `save` to something a little clearer? The textbook does mention something like "Use Nouns for Things and Verbs for Actions" too. Would something `contentToBeSaved` work (kinda long though)?